userguide/snippets/pathlengths.rst

```diff
@@ -34,11 +34,27 @@ can also reduce the length of these limits by one or more bytes.
    | Names               |                |                                                                        |
    +---------------------+----------------+------------------------------------------------------------------------+
    | Mounted Filesystem  | 88 bytes       | Mounted filesystem path length (*MNAMELEN*). Longer paths can prevent  |
-   | Paths               |                | a device from being mounted.                                           |
+   | Paths               |                | a device from being mounted.  See below for examples of path length    |
+   |                     |                | calculation.                                                           |
    +---------------------+----------------+------------------------------------------------------------------------+
    | Device Filesystem   | 63 bytes       | `devfs(8)                                                              |
    | Paths               |                | <https://www.freebsd.org/cgi/man.cgi?query=devfs&sektion=8>`__ device  |
    |                     |                | path lengths (*SPECNAMELEN*). Longer paths can prevent a device from   |
    |                     |                | being created.                                                         |
    +---------------------+----------------+------------------------------------------------------------------------+
 
+Care is needed with regard to ZFS snapshots. **If the mounted path length for a snapshot exceeds 88 characters, 
+then the snapshot and its data will be safe but inaccessible until its mounted path length is reduced**. 
+Typically this is rectified by renaming (shortening) the dataset, snapshot, or mountpoint, so that the path length is
+below the 88 byte limit. Note that ZFS snapshots are automatically mounted on demand within hidden :file:`/.zfs/snapshot/`
+directories, and can also be manually mounted using the command line. The 88 byte limit affects both types of mount.
+
+Examples:
+
+- **Automatic mount:** Any time a user attempts to navigate to a snapshot, or search or access its contents, ZFS must temporarily mount the snapshot. Typically the snapshot :file:`mypool/dataset/snap1@snap2` will be mounted at the path
```

Comment:
(github helpfully renders backticks as highlights and removes them.  They are present here, but it is helping.)

ZFS temporarily mounts a snapshot when a the administrative GUI is used to navigate to it. The mountpoint used is a :file:`.zfs` directory just above the snapshot. For example, the snapshot :file:`mypool/dataset/snap1@snap2` is mounted at :file:`/mnt/mypool/dataset/.zfs/snapshot/snap2/`. If this created mountpoint name exceeds 88 characters, the snapshot is not accessible.

Reply:
Hmm.
1) Not just "when the GUI is used to access it" - they can be accessed by CLI or remotely via Samba etc if "hide dot files" isn't enabled.
2) If we can't expect a reader to understand "it" and referents, can we expect _"The mountpoint used is a .zfs directory just above the snapshot"_ to be understandable? I've had a go at rewording to try and combine the best of both and be clearer. If it doesn't work can we iterate and try again?

Reply:
ZFS automatically mounts snapshots when the data needs to be accessed. Browsing to a snapshot through a file browser, via the CLI, or searching data in a snapshot will cause the data to be mounted. For example, the snapshot :file:mypool/dataset/snap1@snap2 is mounted at :file:/mnt/mypool/dataset/.zfs/snapshot/snap2/. Automatically mounted snapshots must still obey the 88 character limit or the snapshot is not accessible.

Reply:
Is this comment on the original, or on the revised text after Warren's review?

Reply:
I was referring directly to the text above. Seems there is the other large commit that is nearly the same as this one. I'll have a read through the other commit as it sounds like you've fixed most of the issues addressed here.